src/components/sidebar/types.ts

```diff
@@ -127,6 +128,7 @@ interface MenuItem {
 	 * icon will be shown instead.
 	 */
 	trailingIcon?: ReactElement
+	badge?: SidebarNavMenuItemBadgeProps
```

Comment:
This seemed like the simplest fix for this type issue? I do see a few @TS-fixme's related to the badge stuff with the menu items, so a more permanent adjustment might be better.